### PR TITLE
Update default push service endpoint [ci skip]

### PR DESCRIPTION
**Pull request by jonalmeida**
Does this need a changelog entry?

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `cargo test --all` produces no test failures
  - `cargo clippy --all --all-targets --all-features` runs without emitting any warnings
  - `cargo fmt` does not produce any changes to the code
  - `./gradlew ktlint detekt` runs without emitting any warnings
  - `swiftformat --swiftversion 4 megazords components/*/ios && swiftlint` runs without emitting any warnings or producing changes
  - Note: For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/master/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

**Comment by eoger** [comment]
> Does this need a changelog entry?
Yeah even though you're the only user we should still do it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# v0.36.0
+
+## Push
+
+* Updated the default server host for the push service to match the production server.
```
Comment (eoger) [inline CHANGELOG.md]: We don't edit `CHANGELOG.md` directly anymore, do you mind pushing these changes to `CHANGES_UNRELEASED.md`?

Comment (jonalmeida) [reply CHANGELOG.md]: Oops sorry, didn't know!

**Comment by eoger** [comment]
Looks like closing #1430 closed this PR too, sorry.

**Pull request merged**